### PR TITLE
Document almost all of menu.c

### PR DESCRIPTION
**Pull request by FazanaJ**
At the low low cost of my own sanity.

I found homes for a good number of strings too, though some are still a little ambiguous

### CODE REVIEW

**Review by Ryan-Myers — COMMENTED** [review]
Quick thing. Can't do a full one right now.``

File: src/menu.c [hunk 1]
```diff
@@ -6492,40 +5645,45 @@ s32 menu_magic_codes_list_loop(s32 updateRate) {
...
-                       CHEAT_ALL_BALLOONS_ARE_RAINBOW));
-        func_8008A8F8(code, 0x40000,
-                      (CHEAT_ALL_BALLOONS_ARE_RED | CHEAT_ALL_BALLOONS_ARE_GREEN | CHEAT_ALL_BALLOONS_ARE_BLUE |
-                       CHEAT_ALL_BALLOONS_ARE_RAINBOW));
-        func_8008A8F8(code, 0x80000,
-                      (CHEAT_ALL_BALLOONS_ARE_RED | CHEAT_ALL_BALLOONS_ARE_GREEN | CHEAT_ALL_BALLOONS_ARE_BLUE |
-                       CHEAT_ALL_BALLOONS_ARE_YELLOW));
+        code = 1 << gUnlockedCheatIDs[gOptionsMenuItemIndex];
+        gActiveMagicCodes ^= code;                                               // Toggle active cheats?
+        cheatlist_exclusive(code, CHEAT_BIG_CHARACTERS, CHEAT_SMALL_CHARACTERS); // cheatlist_exclusive() = Clear flags?
```
Comment (DavidSM64) [inline src/menu.c:5658]:
```suggestion
        gActiveMagicCodes ^= code;
        cheatlist_exclusive(code, CHEAT_BIG_CHARACTERS, CHEAT_SMALL_CHARACTERS);
```

**Review by DavidSM64 — COMMENTED** [review]
Minor comment edits

**Review by DavidSM64 — APPROVED** [review]
LGTM

**Pull request merged**